pkg/storage/utils/decomposedfs/xattrs/xattrs.go

```diff
@@ -207,6 +225,21 @@ func Set(filePath string, key string, val string) (err error) {
 	return xattr.Set(filePath, key, []byte(val))
 }
 
+// SetWithLock an extended attribute key to the given value with an existing lock
+func SetWithLock(filePath string, key string, val string, fileLock *flock.Flock) (err error) {
+	// check the file is write locked
+	switch {
+	case fileLock == nil:
+		return errors.New("no lock provided")
+	case fileLock.Path() != filePath+".flock":
```

Comment:
There is a func to get the lock file name for a file - should be used here. This cancatenation is not nice.